### DISCUSSION TITLE
Eseguire routine Alexa tramite sensore di movimento su Hassio

### DISCUSSION THREAD

**Discussion by paolomorabito** [indomus/forum#1579] [HUB: Home Assistant]
Ho HASSIO collegato ad Alexa tramite haaska. Funziona tutto perfettamente. Tra le altre cose, condivido con Alexa alcuni sensori, tra cui un sensore di movimento (device_class: motion). Il sensore viene riconosciuto da Alexa e posso creare (sullìapp Alexa) una routine che ha come trigger questo sensore. L'azione di questa routine è quella di accendere una lampadina. 

Quando testo la routine dall'app tutto funziona (la lampadina si accende), ma quando il sensore (trigger) cambia lo stato, la routine non parte.  Su Alexa il sensore funziona perfettamente, cioè cambia lo stato quando è presente un movimento, e si ripristina dopo 1 minuto.  Non riesco a capire per quale motivo non parte la routine. 

Qualcuno sa dove sta il problema?

**Comment by KC-inDomus** [comment]
è proprio sbagliato l'approccio. non ha senso integrare home assistent su alexa per poi relegare le automazioni su alexa.

**Comment by paolomorabito** [comment]
Ok, cambio esempio per spiegarmi meglio.

Stessa situazione precedente, solo che invece di accendere una lampadina, voglio far partire la skill di Reolink che mi apre un flusso video sul mio Echo Show, proveniente da una telecamera in giardino (operazione che posso fare SOLO da Alexa, non da HASSIO, pronunciando le parole: "Alexa, mostra telecamera giardino")

Ho creato una routine su Alexa. Se la testo da app Alexa funziona. Non funziona invece in autonomia, cioè quando scatta veramente il trigger.

Ho trovato la soluzione usando Voice Monkey, ma non capisco per quale motivo non debba funzionare con una semplice routine su Alexa.

**Reply by Larry-78** [reply]
Ciao concordo con @KC-inDomus e secondo me nel tuo caso relativo alla cam potresti provare a creare un'automazione utilizzando [Run Custom Command](https://github.com/custom-components/alexa_media_player/wiki#run-custom-command) appartenente al custom component [Alexa Media Player](https://github.com/custom-components/alexa_media_player).
Per fare un esempio

```yaml
service: media_player.play_media
target:
  entity_id: media_player.tuo_Alxa_Echo_Show
data:
  media_content_type: custom
  media_content_id: 'mostra telecamera giardino'
```
In questo modo simuli un tuo comando vocale e Alexa di conseguenza risponderà ed eseguirà la routine.
Oppure ci sarebbe anche la possibilità di operare direttamente con le [Alexa Routines](https://github.com/custom-components/alexa_media_player/wiki#alexa-routines)

**Reply by paolomorabito** [reply]
Grazie, proverò in questo modo. 

Giusto un'ultima domanda: se volessi aprire una specifica pagina web (da mostrare su echo show) utilizzando il custom component suggerito, c'è qualche modo? Tra gli esempi non trovo qualcosa di simile, ma è da tempo che vorrei creare un'automazione del genere direttamente da Hassio.

Grazie ancora

**Reply by paolomorabito** [reply]
Mi rispondo da solo. Pare non ci sia modo. 
O meglio, si potrebbe creare una skill ad hoc che, tramite uso di APL, apra una pagina web sugli echo show

E' quello che ho fatto. E funziona

**Reply by Larry-78** [reply]
Ciao, ho cercato una soluzione ma purtroppo con poco tempo a disposizione, felice di sapere che l'hai trovata, sapevo che con quel custom component non era possibile ma eri già a conoscenza. 
Invece mi sono bloccato con la skill Reolink, ho provato come hai fatto tu ma non c'è verso di aprire flusso video dalle mie RLC-420-5MP, Mi risponde che "..sul dispositivo questa funzione non è disponibile". Tu che modello di cam hai? Se hai una dritta te ne sarei grato
